compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -289,4 +308,97 @@ private void storeDiagnosticInModule(PackageID pkgId, Diagnostic diagnostic) {
         BLangPackage pkgNode = this.packageCache.get(pkgId);
         pkgNode.addDiagnostic(diagnostic);
     }
+
+    private boolean hasOtherTypeInArgs(Object[] args) {
+        BType constraint;
+        BType bType;
+        BTypeSymbol bTypeSymbol;
+        for (Object arg : args) {
+            if (arg instanceof BType) {
+                switch (((BType) arg).getKind()) {
+                    case OTHER:
+                        return true;
+                    case TUPLE:
+                        List<BTupleMember> members = ((BTupleType) arg).getMembers();
+                        for (BTupleMember bTupleMember : members) {
```

Comment:
Let's recursively call `hasOtherTypeInArgs` here for the tuple members, in cases like where the tuple member is a tuple itself.